### PR TITLE
Added support to TIF reader/writer and add electron tomographic data

### PR DESCRIPTION
**Pull request by philouc**
Add a workflow for the pysap-etomo plugin

### CODE REVIEW

**Comment by philouc** [comment]
@sfarrens Could you review this PR?

**Comment by sfarrens** [comment]
> @sfarrens Could you review this PR?

Sure, I will try to have a look at it before the meeting on Friday.

File: pysap/data.py [hunk 1]
```diff
@@ -119,6 +119,11 @@
                 "master/pysap-data/gen.mr"),
         "md5sum": None
     }
```
Comment (sfarrens) [inline pysap/data.py:121]:
```suggestion
    },
```

File: pysap/base/loaders/tif.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    allowed_extensions = [".tif"]
+
+    def load(self, path, as_gray=False):
+        """ A method that load the data in TIF format.
+
+        Parameters
+        ----------
+        path: str
+            the path to the data to be loaded.
+        meta_field: str, default 'metadata'
```
Comment (sfarrens) [inline pysap/base/loaders/tif.py:30]: `meta_field` is not one of the method arguments and `as_gray` is missing.

File: pysap/base/loaders/tif.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+    def save(self, image, outpath, check_contrast=True):
+        """ A method that save the image in TIF format.
+
+        Parameters
+        ----------
+        image: Image
+            the image to be saved.
+        outpath: str
+            the path where the the image will be saved.
```
Comment (sfarrens) [inline pysap/base/loaders/tif.py:52]: Missing parameter for `check_contrast`

File: pysap/base/loaders/tif.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            the loaded image.
+        """
+        _array = imread(path, as_gray)
+        _meta = {"path": path}
+        return Image(data_type="scalar",
+                     metadata=_meta,
+                     data=_array)
+
+    def save(self, image, outpath, check_contrast=True):
+        """ A method that save the image in TIF format.
```
Comment (sfarrens) [inline pysap/base/loaders/tif.py:45]:
```suggestion
        """ A method that saves the image in TIF format.
```

**Review by sfarrens — CHANGES_REQUESTED** [review]

**Comment by sfarrens** [comment]
@philouc the implementation looks good. I pointed out a few small corrections to make. Note that the target branch for PRs should now be `develop`, so please make sure you pull all the latest commits to that branch.

**Comment by sfarrens** [comment]
@philouc just a little reminder that there are some updates to be made (see above) on this PR before it can be merged. 🙂

File: pysap/base/loaders/tif.py (hunk 2 above)
Comment (philouc) [reply pysap/base/loaders/tif.py:30]: I don't understand I can no longer edit the files to fix the above mentioned issue. That's really messy!

**Comment by chaithyagr** [comment]
@philouc I think you need to add scikit-image to the requirements of the package, now that we import it also.

You can add it here:
https://github.com/CEA-COSMIC/pysap/blob/d72aa5284a411f91435232cf977e97cb7d08ec79/pysap/info.py#L69-L80

**Pull request closed without merging**